sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -308,4 +313,48 @@ export class StudentController extends BaseController {
 
     stream.pipe(response);
   }
+
+  @Get("application-summary")
+  async getStudentApplicationSummary(
+    @UserToken() userToken: IUserToken,
+  ): Promise<StudentApplicationDTO[]> {
+    const existingStudent = await this.studentService.getStudentByUserName(
+      userToken.userName,
+    );
+    if (!existingStudent) {
+      throw new NotFoundException(
+        `No student was found with the student name ${userToken.userName}`,
+      );
+    }
+    const application = await this.applicationService.getAllStudentApplications(
+      existingStudent.id,
+    );
+    let response = [] as StudentApplicationDTO[];
+    for await (const element of application) {
```

Comment:
The performance impact of doing it in for loop awaiting for every single item is huge and it will be getting worse as the list gets bigger. As commented below, if we are going to use the offering_id from applications table, I believe that we can resolve all that is needed is a DB query.

Reply:
Please change the element to a more meaningful name.

Reply:
we are not going to load the whole data at once, as it will cause performance issues, we will be using pagination with page calls?